CockpitPlugin/src/main/java/com/polidea/cockpitplugin/input/InputFilesProvider.kt

```diff
@@ -6,39 +6,140 @@ import java.io.File
+        //
+        // Someone could ask a question: "Why `staging` flavor is first in variant's name `stagingDemoDebug`?
+        // Why the variant name is `stagingDemoDebug` not `demoStagingDebug`?" The name (and flavors' priority) is determined by
+        // order of flavorDimensions in build.gradle file. In our case this is:
+        //
+        //        flavorDimensions "api", "mode"
+        //
+        // `staging` belongs to `api` dimension, so it's the first
+        // `demo` belongs to `mode` dimesion, so it's the second in `stagingDemoDebug` variant's name
+        //
+        // Finally, resource merging priority is following:
+        // main < demo < staging < demoStaging < debug < stagingDemoDebug
+        //
+        // It corresponds to Cockpit files priority:
+        //
+        // cockpit.yml < cockpitDemo.yml < cockpitStaging.yml < cockpitDemoStaging.yml < cockpitDebug.yml < cockpitStagingDemoDebug.yml
+        //
+        // Let's find out how to create that list of files:
+        // [cockpit.yml, cockpitDemo.yml, cockpitStaging.yml, cockpitDemoStaging.yml, cockpitDebug.yml, cockpitStagingDemoDebug.yml]
+        //
 
-        val ordered = ArrayList<String>()
-        ordered.add("") // main
-        val singleFlavors = dimensions.reversed().map { dimension ->
+        val variantsOrderedFromLowestPriority = ArrayList<String>()
+
+        // `main` has the lowest priority, so we add it as a first
+        // Empty string corresponds with `main` variant
+        // because cockpit.yml doesn't contains "main" string in its name
+        variantsOrderedFromLowestPriority.add("")
+
+        // Next, the lowest priority has `demo` and `staging` flavors.
+        //
+        // Declared dimension's order is following:
+        //
+        //        flavorDimensions "api", "mode"
+        //
+        // so we have to reverse dimensions list, to add `demo` (from `mode` dimension) first
+        // and `staging` (from `api` dimension) after `demo`
+        //
+        // Only `staging` and `demo` flavors should be considered, because only those two flavors appear in
+        // variant `stagingDemoDebug`. So we find only those variants which appears in variant's name.
+
+        val singleFlavors = dimensions.reversed().mapNotNull { dimension ->
             flavors.find { it.dimension == dimension && variantName.contains(it.name, ignoreCase = true) }?.name
-        }.filter { it != null }
-                .map { it!!.capitalize() } // Demo, Staging
+        }.map { it.capitalize() } // Demo, Staging
 
-        ordered.addAll(singleFlavors)
+        variantsOrderedFromLowestPriority.addAll(singleFlavors)
 
+        // Composite flavor exists only when application has more than 1 flavor dimension
         if (dimensions.size > 1) {
+
+            // Next, the lowest priority has `stagingDemo` flavor
+            //
+            // singleFlavors is equals ["Demo", "Staging"], so we have to reverse again the list
```

Comment:
It's either "is equal" or "equals". ;)

Also please rephrase: "(...) so we have to reverse again the list" to "(...) so we have to reverse the list again"